src/parser/shared/modules/BaseSelfHealing.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import Events from 'parser/core/Events';
+import { formatThousands, formatPercentage } from 'common/format';
+import HealingDone from 'parser/shared/modules/HealingDone';
+
+/*
+ * Base class for self healing calculations. Handles rendering and calculations, each spec 
+ * has a class that inherits from this to grab spec specific spell IDs. I.E. HolyPaladinSelfHealing
+ */
+
+class BaseSelfHealing extends Analyzer {
+    static dependencies = {
+        healingDone: HealingDone,
+    };
+
+    _healing = 0;
+    _overhealing = 0;
+    _selfHealPercent = 0;
+    static SPELL_ARRAY = [];
+
+    constructor(...args) {
+        super(...args);
+        this.addEventListener(Events.heal.by(SELECTED_PLAYER).to(SELECTED_PLAYER).spell(this.constructor.SPELL_ARRAY), this._onHeal);
+    }
+
+    _onHeal(event) {
```

Comment:
Should there be a check to only penalize self healing below a threshold? i.e. If the healer is healing themselves when they are at 10% HP that is often correct, but if it's 90% then it's probably not correct.

Reply:
It's more subjective, I think. If the raid is "healthy", but the next phase of AoE damage is about to start, casting Riptide ahead of time on a few targets to trigger the buffs are important. A tank is always a good option, but if there is only one, sometimes it's just easier to trigger it on yourself. Another case that comes to mind, when a raid is teetering and there are only a few people left, you maybe be healing yourself and a few others quite often compared to earlier in the encounter. It would be diffiicult to classify that action as bad. Perhaps the mechanism could be smart enough to count self-heals when there are other viable healing targets. In any event, I was surfing around at the open "first issues" and wanted to make a comment even though this has been dormant for quite some time.